src/client/datascience/notebook/vscodeNotebookController.ts

```diff
@@ -292,21 +298,32 @@ export class VSCodeNotebookController implements Disposable {
         // If this NotebookController was selected, fire off the event
         this._onNotebookControllerSelected.fire({ notebook: event.notebook, controller: this });
         this._onNotebookControllerSelectionChanged.fire();
-
-        if (pendingCells?.length) {
-            await this.handleExecution([...pendingCells], event.notebook);
```

Comment:
Removed this code, now we handle the `onShouldRunCells` as that's more concrete than using some static variables.